hoot-core/src/main/cpp/hoot/core/io/OsmJsonReader.cpp

```diff
@@ -446,5 +442,120 @@ void OsmJsonReader::scrubBigInts(QString &jsonStr)
   jsonStr.replace(rx2, "\\1\"\\2\"\\3");
 }
 
+void OsmJsonReader::_readFromHttp()
+{
+  if (!_url.isValid())
+    throw HootException("Invalid URL: " + _url.toString());
+  //  Update the `srsname` parameter to use EPSG:4326
+  if (_url.hasQueryItem("srsname"))
+  {
+    _url.removeQueryItem("srsname");
+    _url.addQueryItem("srsname", "EPSG:4326");
+  }
+  bool split = false;
+  int numSplits = 1;
+  vector<thread> threads;
+  //  Check if there is a bounding box
+  if (_url.hasQueryItem("bbox") && _runParallel)
+  {
+    QStringList bbox = _url.allQueryItemValues("bbox");
+    //  Parse the bounding box
+    geos::geom::Envelope envelope = GeometryUtils::envelopeFromConfigString(bbox.last());
+    //  Check if the bounding box needs to be split
+    int lon_div = 1;
+    int lat_div = 1;
+    //  Don't split an envelope if it is just a little bigger than the prescribed max
+    if (envelope.getWidth() > _coordGridSize * 1.5)
+      lon_div = (int)std::ceil(envelope.getWidth() / _coordGridSize);
+    if (envelope.getHeight() > _coordGridSize * 1.5)
+      lat_div = (int)std::ceil(envelope.getHeight() / _coordGridSize);
+    numSplits = lat_div * lon_div;
+    //  Create envelopes for splitting the request
+    if (lon_div != 1 || lat_div != 1)
+    {
+      //  Only spin up enough threads for the work up to the max
+      int max_threads = _threadCount;
+      if (numSplits < max_threads)
+        max_threads = numSplits;
+      //  Fire up the worker threads
+      _bboxContinue = true;
+      for (int i = 0; i < _threadCount; ++i)
```

Comment:
I'm reading this code... and I think that either this loop should be `for i < max_threads` OR max_threads should be assigned to thread_count.

Reply:
Good catch.